### ISSUE TITLE
AsyncTelegram2 hangs the uC, when "unable to connect to server"

### ISSUE THREAD

**Issue by miguelfernando55** [cotestatnt/AsyncTelegram2#118]
When it is not connected to the server, the uC hangs. It also should be async.

**Comment by cotestatnt** [comment]
Hi @miguelfernando55 .
Which microcontroller is it?

**Comment by miguelfernando55** [comment]
esp32-s2-saola.
Here what it is happenning 
![image](https://github.com/cotestatnt/AsyncTelegram2/assets/26027412/f2a79bf3-a0f3-4d9b-92b8-a2e5a294f0db)
When it is unable, it delays the normal functionality of the rest of the system.
As you may check in the image, I have an interval to send new info. When it happens "unable ... ", it is much longer

**Comment by cotestatnt** [comment]
Thanks for your feedback, I will try to check after 20th August because I am going on vacation from next weekend

**Comment by miguelfernando55** [comment]
Ok. Nice holidays. Hope you may upgrade the functionality.

**Comment by miguelfernando55** [comment]
Is it able to use a flag that warns when the "unable to connect to server" happens?

**Comment by cotestatnt** [comment]
Hi @miguelfernando55 
Regarding your issue, it is not clear to me if the failed connection to the Telegram server happens even if the esp32 is correctly connected to the internet.

If you don't have an active internet connection, you should avoid calling the `getNewMessage()` method (and more generally the other library methods) because an [attempt to connect to the server](https://github.com/cotestatnt/AsyncTelegram2/blob/master/src/AsyncTelegram2.cpp#L29) will still be made and this step is blocking (you can set a timeout eventually).

Checking the internet connection cannot be "internal" to the library because it is strictly linked to the hardware interface (WiFi, Ethernet, GPRS etctec).

**Comment by miguelfernando55** [comment]
You are right. The esp32 may be connected to the router, but the router lost internet connection. It was because of this situation that the "unable to ...." happenned.
I am blocking and not able to design a way to avoid this situation.

**Comment by miguelfernando55** [comment]
Are you able to provide me a status variable when the connection is hanging?
